### DISCUSSION TITLE
Good first issues

### DISCUSSION THREAD

**Discussion by saulshanabrook** [saulshanabrook/jupyterlab-accessibility#7] [General]
When we last chatted, @rodylipson asked where a good place to start might be, if people wanted to start working on a task. I said I would collect a list of good first issues that people could work on and also write down some steps on how to start working on them.

## Finding Issues

Generally speaking, anyone can work on any issue. It's kind of a first come first serve situation. If you are starting to work on you, you can post on the issue saying something like "Hey I am starting to look into this issue" just to let other people know, so someone else doesn't as well. But that isn't required. Also if you start working on one, and hit any problems, feel free to post a comment on the issue saying something like "I am confused about this part, could someone clarify what they mean xxx" or whatever.

 There are always more issues than people to solve then, so it's just a matter of finding on that seems interesting and not too confusing!

Issues are labeled with tags to help people find relevant ones. One tag that is good to look under is ["good first issue"](https://github.com/jupyterlab/jupyterlab/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22). This means someone thought it might be a nice place for someone new to start. The other is ["accessibility"](https://github.com/jupyterlab/jupyterlab/issues?q=is%3Aissue+is%3Aopen++label%3Atag%3AAccessibility+), which are the ones that people added related to accessibility. No one is really forced to tag issues, and they aren't always consistent, so feel free to also ignore them (see an issue I opened a long time ago about doing this more consistently! https://github.com/jupyterlab/jupyterlab/issues/4471)

Here are some I found, but feel free to use any:

* https://github.com/jupyterlab/jupyterlab/issues/9874: Seems like the easiest one! Literally maybe just a find and replace of ... (three dots) for … (ellipsis character)! Someone should def grab this one :)
* https://github.com/jupyterlab/jupyterlab/issues/9686: The "easy part" of this seems doable. It involves adding labels (possibly with this [`aria-label` tag](https://www.w3.org/WAI/tutorials/forms/labels/#using-aria-label)) to the sidebar and `tabindex="0"`. Might require changes in lumino
* https://github.com/jupyterlab/jupyterlab/issues/9684
* https://github.com/jupyterlab/jupyterlab/issues/9682 adding labels to other elements
* https://github.com/jupyterlab/jupyterlab/issues/9491 changing roles of different parts. There are many different smaller issues in here. If folks just wanted to tackle on to begin with, that's fine! Even just a one line fix could be good to get started, adding these `role` tags.
* https://github.com/jupyterlab/jupyterlab/issues/8832 If you really love colors you might like this one. @isabela-pf has done a ton of research here, but it still seems like there is more to be done. Probably would be good to read over the issue and check in with her about what some next steps could be.
* https://github.com/jupyterlab/jupyterlab/issues/6582 this could be not too bad. looks like someone was working on this a while ago, but never finished it: https://github.com/jupyterlab/jupyterlab/pull/6583 it's generally OK to take over someone's fix if they haven't worked on it in a while. it seems like they tried to fix a bunch of other things. But a fix for this might be just looking their fixes, finding the part that just fixes this issue, and making a new pull request with just that fix, and just crediting them in the description like "I pulled the changes from https://github.com/jupyterlab/jupyterlab/pull/6583 thanks @diagram-codesprint for the code!" related issue: https://github.com/jupyterlab/jupyterlab/issues/6577 
* dialogue box issues https://github.com/jupyterlab/jupyterlab/issues/6581 https://github.com/jupyterlab/jupyterlab/issues/6579

Going to post this for now, and then  come back later to add some background on some concepts like "lumino" and "css", as well as a more intro version of the ["Contributing to JupyterLab" post](https://jupyterlab.readthedocs.io/en/stable/developer/contributing.html), but feel free to look at that until then!

**Comment by isabela-pf** [comment]
Yup! [jupyterlab/jupyterlab #8832](https://github.com/jupyterlab/jupyterlab/issues/8832) is a work in progress. I am happy to bring people up to speed with it, especially since I've only found some of the fixes I think need to be made.
